sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -441,7 +442,7 @@ class Analyzer(
 
       // When resolve `SortOrder`s in Sort based on child, don't report errors as
       // we still have chance to resolve it based on grandchild
-      case s @ Sort(ordering, global, child) if child.resolved && !s.resolved =>
```

Comment:
But if we have `order by a` and after resolve the ordering column, we will still hit this case, which is not what we want, right?